gitlab/v4/objects/users.py

```diff
@@ -319,6 +323,9 @@ class UserManager(CRUDMixin, RESTManager):
     )
     _types = {"confirm": types.LowercaseStringAttribute, "avatar": types.ImageAttribute}
 
+    def get(self, id: Union[str, int], lazy: bool = False, **kwargs: Any) -> User:
+        return cast(User, super().get(id=id, lazy=lazy, **kwargs))
```

Comment:
question: I already merged this for projects but now it has me thinking, is there any way to avoid adding all this boilerplate for typing? There will be a _lot_ of these I imagine if as we go module by module, and I guess will need to be copy/pasted for a lot of mixin usage.

Reply:
I have tried and tried again and tried again... But so far I can't figure out a way to do it. If you can figure it out it will be awesome!

There are ways to do it if you return a `self` object. But since we are returning `self._obj_cls()` it doesn't seem like things are smart enough to understand that yet. Well at least not `mypy`

This will help editors and the type-checker in knowing that `gl.users.get()` returns a `User` object. Otherwise it will think it returns a `RESTObject` which isn't as useful.

Reply:
Oh, you will probably like my idea for handling `gl.users.list()` even less 😜

At the moment my best solution for that seems to be adding a `UserList` class or something like that. I haven't figured it out yet but thinking about it.